frontend/src/components/App/AppContainer.tsx

```diff
@@ -31,6 +31,39 @@ window.desktopApi?.receive('backend-token', (token: string) => {
   setBackendToken(token);
 });
 
+/**
+ * QueryParamRedirect is a component that checks for a 'to' query parameter and redirects accordingly
```

Comment:
Could you please add a section to here about when it is used? 
(I assume it's used by backstage somehow? Can you please let me know how?)

The comment in the git commit would be useful here too.
> Example: /notifications?to=/settings/general redirect to /settings/general

Reply:
The Backstage plugin provides an "Open in Headlamp" button.
When clicked, the user is redirected to the Backstage page containing a Headlamp iframe. The iframe URL includes a redirect query parameter (to), which Headlamp uses to navigate the user to the correct page or resource (e.g., a Deployment or other Kubernetes resource). I didn't include this flow in the commit message because this redirect query param can be used by plugins and other places.

In short: clicking "Open in Headlamp" on a resource in Backstage opens that resource directly in Headlamp.